iXBRLViewerPlugin/viewer/src/js/ixbrlviewer.js

```diff
@@ -77,6 +77,41 @@ iXBRLViewer.prototype.pluginPromise = function (methodName, ...args) {
     });
 }
 
+iXBRLViewer.prototype.setFeatures = function(featureNames, queryString) {
+    const iv = this;
+    const featureMap = {}
+    // Enable given features initially
+    featureNames.forEach(f => {
+        featureMap[f] = true;
+    });
+
+    // Enable/disable features based on query string
+    const urlParams = new URLSearchParams(queryString);
+    urlParams.forEach((value, key) => {
+        // Do nothing if feature has already been disabled by query
+        if (featureMap[key] !== false) {
+            // Disable feature if value is exactly 'false', anything else enables the feature
+            featureMap[key] = value !== 'false';
+        }
+    });
+
+    // Gather results in _features set
+    iv._features = new Set();
+    Object.keys(featureMap).forEach(f => {
```

Comment:
nit: could use `Object.entries` and `iv._features` is already initialized to an empty set?

Reply:
I didn't want to introduce the possibility of having `_features` having existing items in it. If later we want to be able to "refresh" enabled features without completely reloading, we'll want this to reset `_features` from scratch. I changed `_features` to initialize to null because we should expect an error if it's ever referenced before `setFeatures` has been called.